File: examples/python/07_proxy_outcome.py
"""Simulated proxy/delayed-primary-outcome CMR example."""

import numpy as np

import cmrdesign as cmr

rng = np.random.default_rng(107)
n_pilot = 100
d = np.r_[np.ones(n_pilot // 2), np.zeros(n_pilot // 2)]
proxy_y = np.r_[
    rng.binomial(1, 0.35, n_pilot // 2),
    rng.binomial(1, 0.25, n_pilot // 2),
]

fit = cmr.cmr_proxy(proxy_y, d, zeta={"treatment": 0.04, "control": 0.06}, method="bernoulli")

print({"pi_treatment": fit.pi, "certificate": fit.U_CMR})
print(fit.confidence_set.extra["bridge"])

File: examples/r/07_proxy_outcome.R
# Simulated proxy/delayed-primary-outcome CMR example.

library(cmrdesign)

set.seed(107)
n_pilot <- 100L
d <- c(rep(1, n_pilot / 2), rep(0, n_pilot / 2))
proxy_y <- c(rbinom(n_pilot / 2, 1, 0.35), rbinom(n_pilot / 2, 1, 0.25))

fit <- cmr_proxy(
  proxy_y,
  d,
  zeta = c(treatment = 0.04, control = 0.06),
  method = "bernoulli"
)

print(list(pi_treatment = fit$pi, certificate = fit$U_CMR))
print(fit$confidence_set$bridge)
